### DISCUSSION TITLE
Windows 0.0.0.0:53 bind error

### DISCUSSION THREAD

**Discussion by c-michail** [mokeyish/smartdns-rs#463] [General]
I am getting on windows with the latest version:
cound not bind to UDP: 0.0.0.0:53, Only one usage of each socket address (protocol/network address/port) is normally permitted. (os error 10048)

**Comment by mokeyish** [comment]
If the port is not already opened, it may be an address reuse issue, but this is not covered in the update.


https://github.com/mokeyish/smartdns-rs/blob/d9b544b4b3beda3f02bdfe1b5c2a90f2f2b0138c/src/server/net.rs#L31

**Reply by c-michail** [reply]
@mokeyish 
According to this article Hyper-V is using the 53 port and can be shared: (I have Hyper-V installed)
https://superuser.com/questions/1633432/why-windows-10-has-a-svchost-process-that-listen-on-port-53

btw you're reuse the address not the port too so I tried to add `sock_ref.set_reuse_port(true)?;`  but  i got an error on compiling:

```
    --> src\server\net.rs:73:12
   |
73 |         sock_ref.set_reuse_port(true)?;
   |                  ^^^^^^^^^^^^^^ method not found in `SockRef<'_>`

```

**Reply by mokeyish** [reply]
It maybe not support on Windows

**Comment by c-michail** [comment]
As you can see the port 53 port isn't used. btw i ran another DNS Server app and it's working.

![εικόνα](https://github.com/user-attachments/assets/53ae30b4-0a96-4339-8dbc-65bf0d83af53)
![εικόνα](https://github.com/user-attachments/assets/beddd9d5-dab2-4fc9-8e73-07a6fd872de3)

**Comment by mokeyish** [comment]
https://help.socketlabs.com/docs/how-to-fix-error-only-one-usage-of-each-socket-address-protocolnetwork-addressport-is-normally-permitted

**Reply by c-michail** [reply]
Again i can use other windows dns servers without problems and also previous versions of smartdns worked.
Seems some depedancy upgrade or change caused this.

**Comment by c-michail** [comment]
Seems the latest release fixed this

**Reply by mokeyish** [reply]
OK, that should be closed.
